### PR TITLE
fix(clustermanager): preserve existing annotations when patching EgressIP mark

### PR DESCRIPTION
**Pull request by smulje**
## 📑 Description

The EgressIP mark annotation patch used a JSON Patch `add` operation targeting
`/metadata/annotations` directly. Per RFC 6902, an `add` on an existing object
member replaces its value entirely, so this wiped out any other annotations
already present on the EgressIP object (e.g. annotations added by GitOps
tools such as ArgoCD) every time the controller reconciled the object.

This PR targets the specific annotation leaf
(`/metadata/annotations/k8s.ovn.org~1egressip-mark`) instead, only creating
the parent annotations map when it doesn't already exist — mirroring the
existing per-key JSON patch pattern used for pod annotations in `pkg/kube`.
When the annotations map does need to be created, the patch also carries a
`test` on `resourceVersion` so a concurrent writer adding the object's first
annotation can't be silently overwritten; the existing reconcile-retry path
picks it up with fresh state instead.

Fixes #

## Release Notes

```release-note
Fixed a bug where the EgressIP controller could overwrite other
annotations (e.g. ones added by GitOps tooling) on an EgressIP object
while adding its own mark annotation.
```

## Additional Information for reviewers

- `generateMarkPatchOps` now builds the JSON patch ops for the mark
  annotation instead of the old `generateMarkPatchOp`, targeting the
  specific annotation key rather than the whole annotations map.
- `generateEgressIPPatches` and its callers now also thread through the
  object's `resourceVersion` so the guard above can be applied.
- Unit tests in `pkg/clustermanager/egressip_controller_test.go` cover the
  `resourceVersion` guard's boundary conditions (stale vs. matching
  resourceVersion, guard skipped when resourceVersion is unknown) by
  applying real JSON Patch requests through `egressipfake.NewClientset`
  (introduced in #6743), rather than a hand-rolled simulation of RFC 6902
  semantics.
- Added an e2e test in `test/e2e/egressip.go` (`Should preserve a
  pre-existing foreign annotation when the egressip-mark annotation is
  added`) that creates a real EgressIP with a foreign annotation against a
  live cluster and verifies it survives mark assignment.
- Reproduced and verified against a live OCP 4.22 cluster: creating an
  EgressIP with a custom annotation and forcing an ovnkube-control-plane
  resync previously dropped the custom annotation when the mark was
  applied; with this fix the custom annotation is preserved alongside the
  mark.

## ✅ Checks
- [x] My code does not require changes to the documentation
- [x] My code requires tests
- [x] if so, I have added and/or updated the tests as required
- [x] All the tests have passed in the CI

## How to verify it

- Unit tests: `TestGenerateMarkPatchOps*` and `TestEscapeJSONPatchPathKey`
  in `pkg/clustermanager`.
- E2E: the new `Should preserve a pre-existing foreign annotation when the
  egressip-mark annotation is added` test in `test/e2e/egressip.go`.
- Manually: create an EgressIP with a custom annotation but no mark yet,
  then trigger a resync (e.g. restart `ovnkube-control-plane`); the custom
  annotation should remain alongside the newly-added
  `k8s.ovn.org/egressip-mark` annotation.

### CODE REVIEW

File: go-controller/pkg/clustermanager/egressip_controller.go [hunk 1]
```diff
@@ -1815,24 +1815,61 @@ func (eIPC *egressIPClusterController) patchEgressIP(name string, patches ...jso
...
+		// The "add" op below requires the parent object to exist.
+		ops = append(ops, jsonPatchOperation{
+			Operation: "add",
+			Path:      "/metadata/annotations",
+			Value:     map[string]string{},
+		})
 	}
+	ops = append(ops, jsonPatchOperation{
+		Operation: "add",
+		Path:      "/metadata/annotations/" + escapeJSONPatchPathKey(util.EgressIPMarkAnnotation),
```
[the remark this answers is not in the document]
Comment (smulje) [reply go-controller/pkg/clustermanager/egressip_controller.go:1862]: Addressed via a real e2e test in `test/e2e/egressip.go` ("Should preserve a pre-existing foreign annotation when the egressip-mark annotation is added"), which creates an EgressIP with a foreign annotation against a live cluster and verifies it survives mark assignment, rather than relying only on the fake clientset.

[the remark this answers is not in the document]
Comment (smulje) [reply go-controller/pkg/clustermanager/egressip_controller.go:1862]: @Coderabbitai Review thread resolved but can you also approve the changes
